Readme.md

```diff
@@ -232,39 +232,61 @@ the same syntax for flags as the `option` method. The version flags can be named
 program.version('0.0.1', '-v, --vers', 'output the current version');
 ```
 
-## Command-specific options
+## Commands
 
-You can attach options to a command.
+You can specify (sub)commands for your top-level command using `.command`. There are two ways these can be implemented: using an action handler attached to the command, or as a separate executable file (described in more detail later). In the first parameter to `.command` you specify the command name and any command arguments. The arguments may be `<required>` or `[optional]`, and the last argument may also be `variadic...`.
 
-```js
-#!/usr/bin/env node
-
-var program = require('commander');
+For example:
 
+```js
+// Command implemented using action handler (description is supplied separately to `.command`)
+// Returns new command for configuring.
 program
-  .command('rm <dir>')
-  .option('-r, --recursive', 'Remove recursively')
-  .action(function (dir, cmd) {
-    console.log('remove ' + dir + (cmd.recursive ? ' recursively' : ''))
-  })
+  .command('clone <source> [destination]')
+  .description('clone a repository into a newly created directory')
+  .action((source, destination) => {
+    console.log('clone command called');
+  });
 
-program.parse(process.argv)
+// Command implemented using separate executable file (description is second parameter to `.command`)
+// Returns top-level command for adding more commands.
+program
+  .command('start <service>', 'start named service')
+  .command('stop [service]', 'stop named serice, or all if no name supplied');
 ```
 
-A command's options are validated when the command is used. Any unknown options will be reported as an error. However, if an action-based command does not define an action, then the options are not validated.
+### Specify the argument syntax
 
-## Variadic arguments
+You use `.arguments` to specify the arguments for the top-level command, and for subcommands they are included in the `.command` call. Angled brackets (e.g. `<required>`) indicate required input. Square brackets (e.g. `[optional]`) indicate optional input.
 
- The last argument of a command can be variadic, and only the last argument.  To make an argument variadic you have to
- append `...` to the argument name.  Here is an example:
 
 ```js
-#!/usr/bin/env node
+var program = require('commander');
 
-/**
- * Module dependencies.
- */
+program
+  .version('0.1.0')
+  .arguments('<cmd> [env]')
+  .action(function (cmd, env) {
+     cmdValue = cmd;
+     envValue = env;
+  });
+
+program.parse(process.argv);
+
+if (typeof cmdValue === 'undefined') {
+   console.error('no command given!');
+   process.exit(1);
+}
+console.log('command:', cmdValue);
+console.log('environment:', envValue || "no environment given");
+```
+
+### Variadic arguments
```

Comment:
Remove heading for "Variadic arguments", just cover as another example of "Specify the argument syntax"